src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDecreaseInheritedMemberVisibilityTests.cs

```diff
@@ -12,12 +10,295 @@ public class DoNotDecreaseInheritedMemberVisibilityTests : DiagnosticAnalyzerTes
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
+        }
+
+        [Fact]
+        public void DecreaseCSharpMemberVisibility()
+        {
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    private new int MyProp3;
+}", GetCSharpCA2222RuleNameResultAt(8, 21));
+
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    internal new int MyProp3 { get; set; }
+}", GetCSharpCA2222RuleNameResultAt(8, 22), GetCSharpCA2222RuleNameResultAt(8, 32), GetCSharpCA2222RuleNameResultAt(8, 37));
```

Comment:
The guidance is we want to be as close to FxCop as possible -- without tying ourselves into knots to achieve it. That's why I asked "does FxCop emit all three diagnostics"? If you say "no, it only reports on the accessors", then that's what you should do -- even though it's not the best experience.

We want maximum FxCop compatibility to start with so that

1. It minimizes obstacles to people migrating from FxCop to the Roslyn-based replacements; we don't want people seeing lots of new diagnostics.

2. We want to be able to test by comparing FxCop results to Roslyn results and seeing if they're the same.

In a future release we can improve experience -- for example, in this case, by firing only once, on the property.

And again, if it requires excessive effort -- which we'll decide case by case -- to get perfect compat, then we will relax the requirement as appropriate.